crates/sui/tests/full_node_tests.rs

```diff
@@ -204,3 +249,77 @@ async fn test_full_node_cold_sync() -> Result<(), anyhow::Error> {
 
     Ok(())
 }
+
+/// Call this function to set up a network and a fullnode with subscription enabled.
+/// Pass in an unique port for each test case otherwise they may interfere with one another.
+async fn set_up_subscription(port: u16, swarm: &Swarm) -> Result<(SuiNode, Client), anyhow::Error> {
+    let ws_server_url = format!("127.0.0.1:{}", port);
+    let ws_addr: SocketAddr = ws_server_url.parse().unwrap();
+
+    let mut config = swarm.config().generate_fullnode_config();
+    config.websocket_address = Some(ws_addr);
+
+    let node = SuiNode::start(&config).await?;
+
+    let client = WsClientBuilder::default()
+        .build(&format!("ws://{}", ws_server_url))
+        .await?;
+    Ok((node, client))
+}
+
+#[tokio::test]
+async fn test_full_node_sub_to_move_event_ok() -> Result<(), anyhow::Error> {
+    let (swarm, mut context, _) = setup_network_and_wallet().await?;
+    // Pass in an unique port for each test case otherwise they may interfere with one another.
+    let (node, ws_client) = set_up_subscription(6666, &swarm).await?;
+
+    let params = BTreeMap::<String, Value>::new();
+    let mut sub: Subscription<SuiEvent> = ws_client
+        .subscribe(
+            "sui_subscribeMoveEventsByType",
+            rpc_params!["0x2::devnet_nft::MintNFTEvent", params],
+            // TODO: update unsub function when it's added
+            "foo",
+        )
+        .await
+        .unwrap();
+
+    let (sender, object_id, digest) = emit_move_events(&mut context).await?;
+    wait_for_tx(digest, node.state().clone()).await;
```

Comment:
if you are already waiting for a subscription event below you don't also need to wait for the node to sequence the tx.

Reply:
I mean to force the txn to be sequenced before going into the `timeout` check, otherwise we may need to have either a big timeout or an flaky test case